### ISSUE TITLE
Discussion: Organization vs User namespaces

### ISSUE THREAD

**Issue by ewdurbin** [pypa/warehouse#11165]
Related to #11070

How separate should we treat the namespaces for Organizations vs Users? Should Organizations be able to be created that conflict with existing usernames?

**Comment by pradyunsg** [comment]
> Should Organizations be able to be created that conflict with existing usernames?

I think so. This is how most platforms handle things (GitHub, GitLab, BitBucket, [Nuget](https://docs.microsoft.com/en-us/nuget/nuget-org/organizations-on-nuget-org), [NPM](https://docs.npmjs.com/creating-an-organization)), and many organisations already have usernames registered for what would be their organisation name. For example, if we make organisations into a separate namespace, then Google will still go and squat https://pypi.org/user/google/ to avoid confusion.

Given that the workflow for creating organisations seems to involve approval from an admin (based on https://github.com/pypa/warehouse/issues/11070#issuecomment-1097029641), I think this should be feasible to do.

I'm not a lawyer but, maybe, the PSF should ask one about whether it might need documented policy for usernames/organisation names, similar to project names, for handling conflicts around that.

**Comment by miketheman** [comment]
[meta question]: If this is a discussion, should it be moved to the GitHub Discussions feature instead? (Happy to delete my question once resolved either way.)

**Comment by ewdurbin** [comment]
We've never used that, but I'm not opposed I guess. I didn't even realize that tab existed until you mention it 😅

**Comment by miketheman** [comment]
> > Should Organizations be able to be created that conflict with existing usernames?
> 
> I think so. This is how most platforms handle things (GitHub, GitLab, BitBucket, [Nuget](https://docs.microsoft.com/en-us/nuget/nuget-org/organizations-on-nuget-org), [NPM](https://docs.npmjs.com/creating-an-organization))...

I don't think you can create an Organization on most of those platforms that conflict with an **existing** user. They are also considered top-level URIs, e.g. https://github.com/pypa vs https://github.com/miketheman - so nobody could ever create a conflicting username of `pypa`.

I think if there's a desired workflow to upgrade or "claim" or "take over" an existing, unused username, then I'm 100% in support of that, but that's different than being able to **create** a brand new organization account.
The current proposed flow of having an admin make the decision to approve organizations and handle any conflict is a good place to start, and I'd be curious to know when conflicts in naming arise, their volume and impact on admins.

**Comment by di** [comment]
@pradyunsg I'm a little confused by your comment, you say "I think so" in response to whether Organizations should be able to be created that conflict with existing usernames, but then go on to describe a process where organization names can't conflict with usernames. Can you clarify?

**Comment by pradyunsg** [comment]
I removed a "don't" in all the editting I did on that post. Sorry for the confusion!

**Comment by miketheman** [comment]
Noticed today that the "approval" behavior didn't prevent this from occurring, at least in one case:
https://pypi.org/user/microsoft/ vs https://pypi.org/org/microsoft/
